vm/vm.go

```diff
@@ -78,55 +84,61 @@ func NewVM(wasm []byte) (VM, error) {
 		store,
 		func(caller *wasmtime.Caller, offset int32, length int32) (any, *wasmtime.Trap) {
 			if offset < 0 {
-				return nil, wasmtime.NewTrap(store, fmt.Sprintf("Int: invalid offset: %d", offset))
+				return nil, wasmtime.NewTrap(fmt.Sprintf("Int: invalid offset: %d", offset))
 			}
 
 			if length < 2 {
-				return nil, wasmtime.NewTrap(store, fmt.Sprintf("Int: invalid length: %d", length))
+				return nil, wasmtime.NewTrap(fmt.Sprintf("Int: invalid length: %d", length))
 			}
 
 			mem := caller.GetExport("mem").Memory()
 
-			bytes := C.GoBytes(mem.Data(), C.int(length))
+			bytes := C.GoBytes(mem.Data(store), C.int(length))
 
 			value := new(big.Int).SetBytes(bytes[1:])
 			if bytes[0] == 0 {
 				value = value.Neg(value)
 			}
 
-			return interpreter.NewIntValueFromBigInt(value), nil
+			return interpreter.NewUnmeteredIntValueFromBigInt(value), nil
```

Comment:
why does this (and below) needs to be unmetered?

Reply:
Everything should be metered, just like in the interpreter, but it's WIP code that is not used in production yet, so did not bother with metering yet, just got the code to compile again